test/e2e/trafficcontrol_test.go

```diff
@@ -0,0 +1,334 @@
+	defer data.crdClient.CrdV1alpha2().TrafficControls().Delete(context.TODO(), tc.Name, metav1.DeleteOptions{})
+	// Wait flows of the TrafficControl to be realized.
+	time.Sleep(time.Second)
+
+	// Verify the number of mirrored packets.
+	verifyMirroredPackets(t, data, portName, true)
+}
+
+func verifyRedirectedPackets(t *testing.T, data *TestData, targetPort, returnPort string) {
+	// Get the number of received and sent packets on test Pod before testing redirect.
+	packetsBefore := countPackets(t, data, "eth0", false, tcTestConfig.podName, "RX") +
+		countPackets(t, data, "eth0", false, tcTestConfig.podName, "TX")
+	// Get the number of received packets on target port before testing redirect. Note that, received packets on veth pair
+	// are counted as TX.
+	packetsOnTargetPortBefore := countPackets(t, data, targetPort, true, "", "TX")
+	// Get the number of sent packets on return port before testing redirect. Note that, sent packets on veth pair are
+	// counted as RX.
+	packetsOnReturnPortBefore := countPackets(t, data, returnPort, true, "", "RX")
+
+	for _, ip := range tcTestConfig.podIPs {
+		cmd := fmt.Sprintf("curl --connect-timeout 1 --retry 5 --retry-connrefused http://%s/hostname", net.JoinHostPort(ip, "8080"))
+		t.Logf("Generate packets for redirecting with command '%s'", cmd)
+		for i := 0; i < 10; i++ {
+			hostname, _, err := data.RunCommandFromPod(data.testNamespace, tcTestConfig.collectorPodName, agnhostContainerName, []string{"sh", "-c", cmd})
+			require.NoError(t, err)
+			require.Equal(t, tcTestConfig.podName, hostname)
+		}
+	}
+
+	// Get the number of redirected packets on test Pod.
+	packetsOnPod := countPackets(t, data, "eth0", false, tcTestConfig.podName, "RX") +
+		countPackets(t, data, "eth0", false, tcTestConfig.podName, "TX") - packetsBefore
+	t.Logf("The total number of redirected packets on test Pod is %d", packetsOnPod)
+
+	// Get the number of received packets on target port after testing redirect.
+	packetsOnTargetPort := countPackets(t, data, targetPort, true, "", "TX") - packetsOnTargetPortBefore
+	t.Logf("The actual number of received packets on target port is %d", packetsOnTargetPort)
+
+	// Get the number of sent packets on return port after testing redirect.
+	packetsOnReturnPort := countPackets(t, data, returnPort, true, "", "RX") - packetsOnReturnPortBefore
+	t.Logf("The actual number of sent packets on return port is %d", packetsOnReturnPort)
+
+	// The difference in the number of packets redirected and received should be within 10.
+	require.GreaterOrEqual(t, 10, abs(packetsOnTargetPort, packetsOnPod))
+	require.GreaterOrEqual(t, 10, abs(packetsOnReturnPort, packetsOnPod))
+	require.GreaterOrEqual(t, 10, abs(packetsOnTargetPort, packetsOnReturnPort))
+}
+
+func testRedirectToLocal(t *testing.T, data *TestData) {
+	skipIfProviderIs(t, "kind", "iproute2 is not installed on Kind Node")
```

Comment:
iproute2 is available on kind node in my testbed, did you see error output saying ip link is not available?
anyway you could still use a hostNetwork Pod to create the interfaces like what has done in Egress test and above code that creates the tunnel device?

Reply:
A hostNetwork Pod is used to create veth pair.